daffodil-lib/src/main/scala/org/apache/daffodil/xml/PUARemappers.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.daffodil.xml
+
+import org.apache.daffodil.exceptions.Assert
+import org.apache.daffodil.util.CharacterSetRemapper
+
+object Unicode {
+  def isLeadingSurrogate(c: Char) = {
+    c >= 0xD800 && c <= 0xDBFF
+  }
+
+  def isTrailingSurrogate(c: Char) = {
+    c >= 0xDC00 && c <= 0xDFFF
+  }
+}
```

Comment:
I just noticed that `java.lang.Character` has `isHighSurrogate` and `isLowSurrogate` functions that I think are the same as these functions.